### PR TITLE
refactor: changed MongoDB string to MongoDSN for security

### PR DESCRIPTION
**Pull request by GoLanguage-design**

### CODE REVIEW

**Review by tclasen — CHANGES_REQUESTED** [review]
Please don't use `os.getenv()` to get environment variables. The intended purpose of the pydantic.BaseSettings classes are to model the environment variables needed to run the application, and it has first-class support for MongoDB DSNs: https://docs.pydantic.dev/latest/usage/types/#pydantic-types

**Comment by DoodlePooge** [comment]
Your `TEMPLATE.env` should reflect the changes you made in `.env`, remember that the TEMPLATE is the framework for your actual env.

File: server/darkmoon/settings.py [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
+
+###########
+# CLASSES #
+###########
+
 
 class Settings(BaseSettings):
     """Settings class."""
 
-    DATABASE_URL: str = "mongodb://darkmoon:password@172.16.5.3/"
```
Comment (DoodlePooge) [inline server/darkmoon/settings.py:8]: Backing what Tory said. Also the fields in the Settings class should be a 1 to 1 match from your .env

You can see in the main branch how we have `DATABASE_URL` in '.env` and in our Setting model


Here's their Settings model example: [https://docs.pydantic.dev/latest/usage/settings/](url)

**Review by DoodlePooge — CHANGES_REQUESTED** [review]

**Review by DoodlePooge — DISMISSED** [review]

File: server/darkmoon/settings.py [hunk 2]
```diff
@@ -23,11 +22,11 @@
...
     """Settings class."""
 
-    scheme = str(os.getenv("SCHEME"))
-    ipv4 = str(os.getenv("IP_ADDRESS"))
-    port = str(os.getenv("PORT"))
-    user = str(os.getenv("DATABASE_USERNAME"))
-    password = str(os.getenv("DATABASE_PASSWORD"))
+    SCHEME: str
+    IP_ADDRESS: str
+    PORT: str
```
Comment (tclasen) [inline server/darkmoon/settings.py:27]: If you don't want to use the Pydantic built-in support for MongoDB DSNs, then the `port` field should be `PORT: int` and the IP address should use the python IPAddress built-in object (Pydantic has a special one you can use as well).

**Review by tclasen — CHANGES_REQUESTED** [review]

File: server/darkmoon/settings.py (hunk 2 above)
Comment (GoLanguage-design) [reply server/darkmoon/settings.py]: The changes have been made! I imported the ipaddress library into settings.py and set IP_ADDRESS to IPv4Address type and PORT to int. These changes are also reflected in the TEMPLATE.env and personal .env files.

File: server/darkmoon/settings.py [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
 
 class Settings(BaseSettings):
     """Settings class."""
 
-    DATABASE_URL: str = "mongodb://darkmoon:password@172.16.5.3/"
+    SCHEME: str
+    IP_ADDRESS: IPv4Address
+    PORT: int
+    USERNAME: str
+    PASSWORD: str
```
Comment (DoodlePooge) [inline server/darkmoon/settings.py:31]: pydantic type: SecretStr

Comment (DoodlePooge) [reply server/darkmoon/settings.py:31]: For sensitive information like a password

Comment (GoLanguage-design) [reply server/darkmoon/settings.py]: Password was changed from str --> SecretStr and the program is still working. IP_ADDRESS also became IPv4 and everything after "/?" was removed in order to accommodate ruff; however, everything seems to still run (GET and POST both return 200)

**Review by tclasen — APPROVED** [review]

**Pull request merged**